azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -303,6 +316,9 @@ def run(self, goal_state):
             if requested_version == CURRENT_VERSION:
                 return
 
+            if not self.__check_if_downgrade_is_requested_and_allowed(requested_version):
```

Comment:
shouldn't this call be done from __should_update_agent?

Reply:
should_update_agent primary checking time window to allow updates. I felt adding this new condition confuses. I'm ok to call it there